solutions/src/main/java/ru/sberbank/school/task08/JacksonManager.java

```diff
@@ -0,0 +1,91 @@
+package ru.sberbank.school.task08;
+
+import com.fasterxml.jackson.core.JsonGenerationException;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.util.Solution;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static java.io.File.separator;
+import static ru.sberbank.school.task08.SaveGameException.Type.IO;
+
+/**
+ * 31.05.2019 сериализация с помощью Jackson
+ *
+ * @author Gregory Melnikov
+ */
+
+@Solution(8)
+public class JacksonManager extends SaveGameManager<MapState<GameObject>, GameObject> {
+
+    private ObjectMapper objectMapper;
+
+    public JacksonManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        objectMapper = new ObjectMapper();
+    }
+
+    @Override
+    @SneakyThrows
+    public void saveGame(@NonNull String filename, @NonNull MapState<GameObject> gameState) throws SaveGameException {
+        String path = filesDirectory + separator + filename;
+        try (FileOutputStream fileOutputStream = new FileOutputStream(path)) {
+
+            objectMapper.writeValue(fileOutputStream, gameState);
+
+        } catch (FileNotFoundException | JsonGenerationException | NullPointerException e) {
+            System.err.println(e);
+            throw new SaveGameException("JacksonManager Saving error", e, IO, gameState);
+        } catch (IOException e) {
+            throw new SaveGameException("JacksonManager Saving error", e, IO, gameState);
+        }
+
+    }
+
+    @Override
+    @SneakyThrows
+    @SuppressWarnings("unchecked")
+    public MapState<GameObject> loadGame(@NonNull String filename) throws SaveGameException {
+        String path = filesDirectory + separator + filename;
+        try (FileInputStream fileInputStream = new FileInputStream(path)) {
+
+            MapState<GameObject> saved = objectMapper.readValue(fileInputStream, MapState.class);
+            return saved;
+
+        } catch (FileNotFoundException | ClassCastException | JsonParseException | JsonMappingException | NullPointerException e) {
```

Comment:
Line is longer than 120 characters (found 131).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1352272017>*